src/main/java/org/eclipse/uprotocol/transport/builder/UMessageBuilder.java

```diff
@@ -0,0 +1,315 @@
+/**
+ * SPDX-FileCopyrightText: 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.eclipse.uprotocol.transport.builder;
+
+
+import org.eclipse.uprotocol.v1.UUri;
+import org.eclipse.uprotocol.v1.UMessage;
+import org.eclipse.uprotocol.v1.UAttributes;
+import org.eclipse.uprotocol.v1.UCode;
+import org.eclipse.uprotocol.v1.UPriority;
+import org.eclipse.uprotocol.v1.UMessageType;
+import org.eclipse.uprotocol.v1.UPayloadFormat;
+import org.eclipse.uprotocol.v1.UUID;
+
+import com.google.protobuf.Any;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.Message;
+
+import org.eclipse.uprotocol.uuid.factory.UuidFactory;
+
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Builder for easy construction of the UAttributes object.
+ */
+public class UMessageBuilder {
```

Comment:
this is replacing the protobuf UMessage?

Reply:
No I'm still using uprotocol.v1.UMessage but this builds teh right type of UMessage (request, response, notification, publish).